### PR TITLE
Bug 2100860: Retrieve user-defined  Alertmanager services from shared configmap

### PR DESCRIPTION
**Pull request by simonpasquier**
https://issues.redhat.com/browse/MON-2289

Requires https://github.com/openshift/cluster-monitoring-operator/pull/1690
Required by https://github.com/openshift/console/pull/11712

### CODE REVIEW

File: pkg/console/subresource/consoleserver/types.go
Comment (jhadvig) [inline pkg/console/subresource/consoleserver/types.go:70]:
```suggestion
// MonitoringInfo holds configuration for monitoring related services
```

**Review by jhadvig — APPROVED** [review]
@simonpasquier thanks for the PR! looks good just one nit
/lgtm
/approve

Adding `/hold` label to get the additional approvals, but not sure who is the QE,DOCS/PX contact
/hold

File: pkg/console/subresource/consoleserver/types.go
Comment (jhadvig) [inline pkg/console/subresource/consoleserver/types.go:71]: noticed that [types.go](https://github.com/openshift/console/pull/11712/files#diff-4d2abe7485c7a156905e0fc628a0edf2aaed9ce1a6d0dda5cc7df661fe27d529) already contain some field.. we should have field parity between console and console-operator

**Review by jhadvig — CHANGES_REQUESTED** [review]

File: pkg/console/subresource/consoleserver/types.go
Comment (simonpasquier) [reply pkg/console/subresource/consoleserver/types.go:71]: Correct, we removed the other fields in https://github.com/openshift/console-operator/pull/562 but we never propagate the change to the console code base (https://github.com/openshift/console/pull/9276 was closed on the assumption that the monitoring URLs could be useful eventually for the multi-cluster console story).

Not sure which way we want to go but I see several options:
* Add back the fields in the console-operator project in this PR.
* Remove the fields from the console project as part of https://github.com/openshift/console/pull/11712.
* Resolve the situation once 4.12 opens.

Let me know what you think but I'm happy to take care of it :)

**Comment by spadgett** [comment]
/hold cancel
/retest

Since this now attached to a BZ, the other labels aren't needed.

File: pkg/console/subresource/consoleserver/types.go
Comment (jhadvig) [reply pkg/console/subresource/consoleserver/types.go:71]: @simonpasquier if we need those fields either in console or console-operator, they should be present in both api, so they match. Since I think those fields are still needed we should bring them back part of this PR

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by jhadvig** [comment]
/hold
@simonpasquier you need to update your added unit tests so they contain the `releaseVersion` field in the config.
Check https://github.com/openshift/console-operator/pull/659

**Comment by jhadvig** [comment]
/retest

```
    unmanaged_test.go:18: waiting for cleanup to reach settled state...
--- FAIL: TestEditUnmanagedConfigMap (15.09s)
=== RUN   TestEditUnmanagedService
    unmanaged_test.go:12: waiting for setup to reach settled state...
    unmanaged_test.go:13: changing console operator state to 'Unmanaged'...
    util.go:51: patching Annotation on the console Service
    util.go:58: polling for patched Annotation on the console Service
26 skipped lines...
    unmanaged_test.go:13: changing console operator state to 'Unmanaged'...
    util.go:120: patching MaxUnavailable on the "console" PodDisruptionBudget
    unmanaged_test.go:86: error: poddisruptionbudgets.policy "console" not found
    unmanaged_test.go:18: waiting for cleanup to reach settled state...
    console-operator.go:321: waiting for observed generation 44 to match generation 46... 
--- FAIL: TestEditUnmanagedPodDisruptionBudget (7.26s)
FAIL
FAIL	github.com/openshift/console-operator/test/e2e	111
```

**Pull request merged**
